rasa/core/training/story_reader/yaml_story_reader.py

```diff
@@ -341,31 +384,52 @@ def _parse_checkpoint(self, step: Dict[Text, Any], is_rule_data: bool) -> None:
 
         self._add_checkpoint(checkpoint_name, slots_dict)
 
-    @staticmethod
-    def is_yaml_story_file(file_path: Text) -> bool:
-        """Check if file contains Core training data or rule data in YAML format.
 
-        Args:
-            file_path: Path of the file to check.
+class StoryParser(YAMLStoryReader):
+    """Encapsulate story specific parser behavior."""
 
-        Returns:
-            `True` in case the file is a Core YAML training data or rule data file,
-            `False` otherwise.
-        """
-        suffix = Path(file_path).suffix
+    def _new_part(self, item_name: Text, item: Dict[Text, Any]) -> None:
+        self._new_story_part(item_name, self.source_name)
 
-        if suffix and suffix not in YAML_FILE_EXTENSIONS:
-            return False
+    def _get_item_title(self) -> Text:
+        return KEY_STORY_NAME
 
-        try:
-            content = rasa.utils.io.read_yaml_file(file_path)
-            return any(key in content for key in [KEY_STORIES, KEY_RULES])
-        except Exception as e:
-            # Using broad `Exception` because yaml library is not exposing all Errors
-            common_utils.raise_warning(
-                f"Tried to check if '{file_path}' is a story or rule file, but failed "
-                f"to read it. If this file contains story or rule data, you should "
-                f"investigate this error, otherwise it is probably best to "
-                f"move the file to a different location. Error: {e}"
-            )
-            return False
+    def _get_plural_item_title(self) -> Text:
+        return KEY_STORIES
+
+    def _get_docs_link(self) -> Text:
+        return DOCS_URL_STORIES
+
+
+class RuleParser(YAMLStoryReader):
+    """Encapsulate rule specific parser behavior."""
+
+    def _new_part(self, item_name: Text, item: Dict[Text, Any]) -> None:
+        self._new_rule_part(item_name, self.source_name)
+        conditions = item.get(KEY_RULE_CONDITION, [])
+        self._parse_rule_conditions(conditions)
+        if not item.get(KEY_RULE_APPLIES_ONLY_FOR_CONVERSATION_START):
+            self._parse_rule_snippet_action()
+
+    def _parse_rule_conditions(
+        self, conditions: List[Union[Text, Dict[Text, Any]]]
+    ) -> None:
+        for condition in conditions:
+            self._parse_step(condition)
+
+    def _close_part(self, item: Dict[Text, Any]) -> None:
+        if item.get(KEY_WAIT_FOR_USER_INPUT_AFTER_RULE) is False:
+            self._parse_rule_snippet_action()
+
+    def _get_item_title(self) -> Text:
+        return KEY_RULE_NAME
+
+    def _get_plural_item_title(self) -> Text:
+        return KEY_RULES
+
+    def _get_docs_link(self) -> Text:
+        # TODO
+        return ""
```

Comment:
does this exist by now?

Reply:
will add it in the rule policy documentation pr